### PR TITLE
[tests-only][full-ci]Improve then steps usages inside acceptance tests suite `apiFederationToRoot1`

### PR DESCRIPTION
**Pull request by SagarGi**
## Description
This PR adds `Then` steps to existing API test scenarios to better check the results of `When` steps.

**Suites covered:**
- `apiFederationToRoot1`

## Related Issue

- Part of https://github.com/owncloud/core/issues/39512

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/apiFederationToRoot1/savePublicShare.feature [hunk 1]
```diff
@@ -91,7 +97,8 @@ Feature: Save public shares created by oC users
     And as "Alice" folder "/PARENT" should not exist
 
 
-  Scenario: Mount public share and sharer unshares the share (remote server share)
+  Scenario Outline: Mount public share and sharer unshares the share (remote server share)
+    Given using OCS API version "<ocs_api_version>"
     Given using server "REMOTE"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiFederationToRoot1/savePublicShare.feature:102]:
```suggestion
    And using server "REMOTE"
```

File: tests/acceptance/features/apiFederationToRoot1/savePublicShare.feature [hunk 2]
```diff
@@ -31,17 +31,23 @@ Feature: Save public shares created by oC users
...
+    And user "Brian" has been created with default attributes and without skeleton files
     And user "Alice" has created folder "/PARENT"
     And user "Alice" has created a public link share with settings
       | path        | /PARENT    |
       | permissions | read       |
       | name        | sharedlink |
     And user "Brian" has added the public share created from server "LOCAL" using the sharing API
     When user "Alice" deletes public link share named "sharedlink" in file "/PARENT" using the sharing API
+    Then the OCS status code should be "<http_status_code>"
     Then the HTTP status code should be "200"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiFederationToRoot1/savePublicShare.feature:45]:
```suggestion
    And the HTTP status code should be "200"
```

File: tests/acceptance/features/apiFederationToRoot1/savePublicShare.feature [hunk 3]
```diff
@@ -103,8 +110,15 @@ Feature: Save public shares created by oC users
     And user "Alice" has added the public share created from server "REMOTE" using the sharing API
     And using server "REMOTE"
     When user "RemoteAlice" deletes public link share named "sharedlink" in file "/PARENT" using the sharing API
-    And using server "LOCAL"
-    Then as "Alice" folder "/PARENT" should not exist
+    Then the OCS status code should be "<http_status_code>"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiFederationToRoot1/savePublicShare.feature:113]:
```suggestion
    Then the OCS status code should be "<ocs_status_code>"
```

File: tests/acceptance/features/apiFederationToRoot1/savePublicShare.feature [hunk 4]
```diff
@@ -103,8 +110,15 @@ Feature: Save public shares created by oC users
...
+    And the HTTP status code should be "200"
+    When using server "LOCAL"
+    Then the HTTP status code should be "200"
+    And as "Alice" folder "/PARENT" should not exist
+    Examples:
+      | ocs_api_version | http_status_code |
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiFederationToRoot1/savePublicShare.feature:119]:
```suggestion
      | ocs_api_version | ocs_status_code |
```

File: tests/acceptance/features/apiFederationToRoot1/savePublicShare.feature [hunk 5]
```diff
@@ -31,17 +31,23 @@ Feature: Save public shares created by oC users
...
+    Given using OCS API version "<ocs_api_version>"
+    And user "Brian" has been created with default attributes and without skeleton files
     And user "Alice" has created folder "/PARENT"
     And user "Alice" has created a public link share with settings
       | path        | /PARENT    |
       | permissions | read       |
       | name        | sharedlink |
     And user "Brian" has added the public share created from server "LOCAL" using the sharing API
     When user "Alice" deletes public link share named "sharedlink" in file "/PARENT" using the sharing API
+    Then the OCS status code should be "<http_status_code>"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiFederationToRoot1/savePublicShare.feature:44]:
```suggestion
    Then the OCS status code should be "<ocs_status_code>"
```

File: tests/acceptance/features/apiFederationToRoot1/federated.feature [hunk 6]
```diff
@@ -301,54 +301,58 @@ Feature: federated
...
     And using server "LOCAL"
-    And using OCS API version "<ocs-api-version>"
     When user "Brian" deletes the last federated cloud share with password "invalid" using the sharing API
     Then the OCS status code should be "401"
     And the HTTP status code should be "401"
     And user "Brian" should see the following elements
       | /textfile0.txt |
     When user "Brian" gets the list of federated cloud shares using the sharing API
-    Then the fields of the last response about user "Alice" sharing with user "Brian" should include
+    And the fields of the last response about user "Alice" sharing with user "Brian" should include
```
Comment (SwikritiT) [inline tests/acceptance/features/apiFederationToRoot1/federated.feature:347]: Either this step should be `Then` or the next one should be `And`

Comment (SagarGi) [reply tests/acceptance/features/apiFederationToRoot1/federated.feature:347]: @SwikritiT  this scenario is skiped both on core and ocis and it has a separeate issue feature where status code is checked.

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Review by SwikritiT — COMMENTED** [review]
LGTM

**Review by phil-davis — APPROVED** [review]

**Pull request merged**